test/hotspot/jtreg/runtime/os/TestTransparentHugePageUsage.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) Ampere Computing and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test TestTransparentHugePageUsage
+ * @bug 8315923
+ * @library /test/lib
+ * @requires vm.gc.Parallel & os.family == "linux" & os.maxMemory > 2G
+ * @summary Check if the usage of THP is zero when enabled.
+ * @comment The test is not ParallelGC-specific, but a multi-threaded GC is
+ *          required. So ParallelGC is used here.
+ * @run driver runtime.os.TestTransparentHugePageUsage ${os.processors}
+ */
+
+package runtime.os;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.InputStreamReader;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import jdk.test.lib.process.ProcessTools;
```

Comment:
It almost certainly fails with 512MB huge pages and without madvise when vm_page_size is 64KB, but not likely fails with 2MB huge pages. I would change the check of usage from 0 to 524288 (512MB in KB), and then it would fail with 2MB huge pages.

Reply:
That is a good idea. We don't need perfection, just a test that has a reasonable chance of informing us when someone breaks your logic.